source/src/Archiver.cc

```diff
@@ -41,115 +41,97 @@ namespace dqm4hep {
     }
 
     //-------------------------------------------------------------------------------------------------
 
     Archiver::~Archiver() {
-      if (m_isOpened)
+      if (isOpened())
         close();
     }
 
     //-------------------------------------------------------------------------------------------------
 
-    StatusCode Archiver::open(const std::string &archiveFileName, const std::string &openingMode, bool allowSuffix) {
+    StatusCode Archiver::open(const std::string &fname, const std::string &opMode, bool overwrite) {
       // if already open write the archive if not done
       // and close it before to re-open
-      if (m_isOpened)
+      if (isOpened()) {
         RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, close());
-
-      if (archiveFileName.empty())
+      }
+      if (fname.empty()) {
         return STATUS_CODE_INVALID_PARAMETER;
-
-      if (allowSuffix) {
+      }
+      if (not overwrite) {
         int fileId(0);
-
-        size_t pos = archiveFileName.rfind(".root");
-
+        size_t pos = fname.rfind(".root");
         if (std::string::npos == pos) {
-          dqm_error("Couldn't open archive '{0}' ! Must be a root file !", archiveFileName);
+          dqm_error("Couldn't open archive '{0}' ! Must be a root file !", fname);
           return STATUS_CODE_INVALID_PARAMETER;
         }
-
-        std::string baseArchiveName = archiveFileName.substr(0, pos);
-        std::string fullArchiveName = archiveFileName;
-
+        std::string baseArchiveName = fname.substr(0, pos);
+        std::string fullArchiveName = fname; 
         while (!gSystem->AccessPathName(fullArchiveName.c_str())) {
           std::stringstream ss;
           ss << baseArchiveName << "_" << fileId << ".root";
           fullArchiveName = ss.str();
           fileId++;
         }
-
         m_fileName = fullArchiveName;
-      } else {
-        m_fileName = archiveFileName;
+      } 
+      else {
+        m_fileName = fname;
       }
-
-      m_openingMode = openingMode;
-
+      m_openingMode = opMode;
       dqm_info("Archiver::open: Opening archive {0}", m_fileName);
-
-      m_pArchiveFile = new TFile(m_fileName.c_str(), openingMode.c_str());
-
-      if (nullptr == m_pArchiveFile) {
-        dqm_error("Couldn't open archive '{0}' !", m_fileName);
+      m_file.reset(new TFile(m_fileName.c_str(), m_openingMode.c_str()));
+      if (nullptr == m_file) {
+        dqm_error("Archiver::open: Couldn't open archive '{0}' !", m_fileName);
         return STATUS_CODE_FAILURE;
       }
-
       m_isOpened = true;
-
       return STATUS_CODE_SUCCESS;
     }
 
     //-------------------------------------------------------------------------------------------------
 
     StatusCode Archiver::close() {
-      if (!m_isOpened)
-        return STATUS_CODE_FAILURE;
-
-      m_pArchiveFile->Close();
-      delete m_pArchiveFile;
-
-      m_pArchiveFile = nullptr;
-
+      if (not isOpened()) {
+        return STATUS_CODE_SUCCESS;
```

Comment:
Shouldn't this not be allowed? Trying to close an already closed file might be a sign of a deeper problem somewhere. 
I would suggest to at least output a debug (even a Warning) message.